src/inatinqperf/adaptors/weaviate_adaptor.py

```
@@ -1,257 +1,340 @@
-"""Adaptor for interacting with a running Weaviate instance via HTTP."""
+"""Weaviate adaptor built on top of the official client library.
+
+This adaptor keeps the surface area used by the benchmark intentionally small:
+
+* Schema management is handled through ``client.schema``.
+* Vector ingest happens via ``client.data_object`` calls.
+* Queries rely on the GraphQL endpoint exposed through ``client.query``.
+
+All interactions log high-level intent so that long-running benchmark sessions
+remain debuggable and predictable for developers.
+"""
 
-import json
 import time
 import uuid
 from collections.abc import Sequence
 from http import HTTPStatus
 
 import numpy as np
-import requests
+from datasets import Dataset as HuggingFaceDataset
 from loguru import logger
-from requests import Response, Session
+from weaviate import Client as WeaviateClient
 
-from inatinqperf.adaptors.base import DataPoint, HuggingFaceDataset, Query, SearchResult, VectorDatabase
+from inatinqperf.adaptors.base import DataPoint, Query, SearchResult, VectorDatabase
 from inatinqperf.adaptors.metric import Metric
 
 
 class WeaviateError(RuntimeError):
-    """Raised when Weaviate responds with an unexpected status code."""
+    """Raised when Weaviate responds with an unexpected status code or payload."""
 
 
 class Weaviate(VectorDatabase):
-    """HTTP-based adaptor that manages a single Weaviate class/collection."""
+    """Light-weight wrapper for managing a single class within a Weaviate instance."""
 
-    INATINQ_WEVIATE_QUERY_DIM = 2
+    _DEFAULT_DESCRIPTION = "Collection managed by iNatInqPerf"
+    _ORIGINAL_ID_FIELD = "originalId"
+    _EXPECTED_VECTOR_RANK = 2
 
     def __init__(
         self,
         dataset: HuggingFaceDataset,
-        metric: str = Metric.COSINE,
-        *,
-        base_url: str = "http://localhost:8080",
+        metric: Metric,
+        url: str,
         class_name: str = "collection_name",
         timeout: float = 10.0,
```

do we need timeout?

---

The timeout protects benchmark runs from stalling on network problems or a misconfigured Weaviate instance. If you need longer waits you can increase the value, but having a bound keeps failures predictable. Failure can happen on every network call (schema create/delete, upsert, search). We want to make sure that it fails fast instead of hanging indefinitely when the server is slow or unreachable.